File: cpp_results/cpp_source/question-14-pass-0.cpp
// Construct longest possible sequence of unique elements with given LCM. 
 #include <bits/stdc++.h> 
 using namespace std ;
 void constructArrayWithGivenLCM ( int N ) {
 
     int lcm = 1;
     for ( int i = 1; i < N; i++ ) {
         lcm = lcm * i / __gcd ( lcm, i );
     }
 
     int count = 0;
     int j = 1;
     while ( count < N ) {
         if ( lcm % j == 0 ) {
             cout << j << " ";
             count++;
         }
         j++;
     }
 }
 
 int main () {
     int N;
     cin >> N;
     constructArrayWithGivenLCM ( N );
     return 0;
 }
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 

File: results/preliminary_analysis/baseline_py_results/py_source/question-14-pass-0.py
# Construct longest possible sequence of unique elements with given LCM. 
from math import sqrt, ceil, floor


def constructArrayWithGivenLCM(N): 
    # Find LCM of 1 to N 
    lcm = 1
    for i in range(1, N + 1): 
        lcm = (lcm * i) // gcd(lcm, i) 
  
    # Construct array 
    arr = [] 
    for i in range(1, N + 1): 
        arr.append(i) 
  
    # Repeat elements of array to get LCM 
    for i in range(N, lcm): 
        arr.append(arr[i % N]) 
  
    return arr 


def gcd(a, b): 
    if a == 0: 
        return b 
    return gcd(b % a, a) 


# Driver code 
N = 10
print("LCM of 1 to", N, "is", constructArrayWithGivenLCM(N)[-1]) 


# Output:
# LCM of 1 to 10 is 360